StandaloneMmPkg/Library/StandaloneMmServicesTableLib/StandaloneMmServicesTableLibCore.c

```diff
@@ -0,0 +1,13 @@
+/** @file
+  MM Services Table Library.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <PiMm.h>
+#include <Library/MmServicesTableLib.h>
+#include <Library/DebugLib.h>
+
+EFI_MM_SYSTEM_TABLE  *gMmst = NULL;

```

Comment:
Since gMmst is owned by StandaloneMmServicesTableLibCore instance, should it be initialized by the same lib instance as well?

The patch keeps the gMmst initialization in the StandaloneMmCoreMemoryAllocationLib, which is a bit tricky.

Reply:
@niruiyu, please let us know if this conversation can be resolved. We'd like to get this change in soon if possible.

Reply:
@makubacki, sorry I was on vacation for the last 2 weeks.
Is there any opportunity to avoid tricky implementation?

Today ProcessLibraryConstructorList() is called from StandaloneMmMain() in the very early location.
Is it possible to resolve your concern if moving the call to a later location?